contracts/lease/src/position/mod.rs

```diff
@@ -7,62 +7,45 @@ use finance::{
 };
 
 use crate::{
-    api::{LeaseCoin, PositionSpecDTO},
+    api::LeaseCoin,
     error::{ContractError, ContractResult},
 };
 
 pub use dto::PositionDTO;
+pub use position_spec::PositionSpec;
 pub use status::{Cause, Liquidation, Status};
 
 mod dto;
+mod position_spec;
 mod status;
 
 #[cfg_attr(test, derive(Debug))]
 pub struct Position<Asset, Lpn> {
     amount: Coin<Asset>,
-    liability: Liability,
-    min_asset: Coin<Lpn>,
-    min_sell_asset: Coin<Lpn>,
+    spec: PositionSpec<Lpn>,
 }
 
 impl<Asset, Lpn> Position<Asset, Lpn>
 where
     Asset: Currency,
     Lpn: Currency,
 {
-    fn new_internal(
-        amount: Coin<Asset>,
-        liability: Liability,
-        min_asset: Coin<Lpn>,
-        min_sell_asset: Coin<Lpn>,
-    ) -> Self {
-        let obj = Self {
-            amount,
-            liability,
-            min_asset,
-            min_sell_asset,
-        };
+    fn new_internal(amount: Coin<Asset>, spec: PositionSpec<Lpn>) -> Self {
+        debug_assert!(!spec.min_asset().is_zero());
+        debug_assert!(!spec.min_sell_asset().is_zero());
```

Comment:
the idea of invariant is to guarantee certain properties of the objects of that type ... so let's leverage it and stop checking these objects ... moreover, that might not be possible due to encapsulation